tests/integ/modin/frame/test_align.py

```diff
@@ -0,0 +1,184 @@
+    )
+    native_other_df = native_pd.DataFrame(
+        [[10, 20, 30, 40], [60, 70, 80, 90], [600, 700, 800, 900]],
+        columns=["A", "B", "C", "D"],
+    )
+    native_left, native_right = native_df.align(
+        native_other_df,
+        join=join,
+        axis=0,
+        limit=None,
+        fill_axis=0,
+        broadcast_axis=None,
+    )
+    df = pd.DataFrame(native_df)
+    other_df = pd.DataFrame(native_other_df)
+    left, right = df.align(other_df, join=join, axis=0)
+    assert_frame_equal(left, native_left)
+    assert_frame_equal(right, native_right)
+
+
+@sql_count_checker(query_count=2, join_count=2)
+@pytest.mark.parametrize("join", ["outer", "inner", "left", "right"])
+def test_align_basic_with_nulls_axis0(join):
+    native_df = native_pd.DataFrame(
+        [[1, 2, np.nan, 4], [6, 7, 8, 9]], columns=["D", "B", "E", "A"]
+    )
+    native_other_df = native_pd.DataFrame(
+        [[10, 20, 30, np.nan], [60, np.nan, 80, 90], [600, 700, 800, 900]],
+        columns=["A", "B", "C", "D"],
+    )
+    native_left, native_right = native_df.align(
+        native_other_df,
+        join=join,
+        axis=0,
+        limit=None,
+        fill_axis=0,
+        broadcast_axis=None,
+    )
+    df = pd.DataFrame(native_df)
+    other_df = pd.DataFrame(native_other_df)
+    left, right = df.align(other_df, join=join, axis=0)
+    assert_frame_equal(left, native_left)
+    assert_frame_equal(right, native_right)
+
+
+@sql_count_checker(query_count=2, join_count=2)
+@pytest.mark.parametrize("join", ["outer", "inner", "left", "right"])
+def test_align_basic_with_all_null_col_axis0(join):
+    native_df = native_pd.DataFrame(
+        [[1, 2, np.nan, 4], [6, 7, 8, 9]], columns=["D", "B", "E", "A"]
+    )
+    native_other_df = native_pd.DataFrame(
+        [[10, 20, 30, np.nan], [60, np.nan, 80, 90], [np.nan, np.nan, np.nan, np.nan]],
+        columns=["A", "B", "C", "D"],
+    )
+    native_left, native_right = native_df.align(
+        native_other_df,
+        join=join,
+        axis=0,
+        limit=None,
+        fill_axis=0,
+        broadcast_axis=None,
+    )
+    df = pd.DataFrame(native_df)
+    other_df = pd.DataFrame(native_other_df)
+    left, right = df.align(other_df, join=join, axis=0)
+    assert_frame_equal(left, native_left)
+    assert_frame_equal(right, native_right)
+
+
+@sql_count_checker(query_count=0)
+def test_align_frame_with_nulls_axis_None_negative():
+    df = pd.DataFrame([[1, 2, np.nan, 4], [6, 7, 8, 9]], columns=["D", "B", "E", "A"])
+    other_df = pd.DataFrame(
+        [[10, 20, 30, np.nan], [60, np.nan, 80, 90], [600, 700, 800, 900]],
+        columns=["A", "B", "C", "D"],
+    )
+    with pytest.raises(
+        NotImplementedError,
+        match="Snowpark pandas 'align' method doesn't support 'axis=None'",
+    ):
+        left, right = df.align(other_df, join="outer", axis=None)
+
+
+@sql_count_checker(query_count=0)
+def test_align_frame_fill_value_negative():
```

Comment:
Can we add more negative tests for the remaining parameters that we do not support?

Reply:
Yes, adding a negative test for the unsupported params